src/adlist.c

```diff
@@ -162,11 +169,20 @@ list *listInsertNode(list *list, listNode *old_node, void *value, int after) {
 }
 
 /* Remove the specified node from the specified list.
- * It's up to the caller to free the private value of the node.
+ * The node is freed but it's up to the caller to free the private value of the node.
```

Comment:
while we are at it maybe we can fix the documentation to elaborate that the value will only be freed in case the list was provided with a free function?

Reply:
will do